File: AndroidSDK/Basic/Lesson_6/Inote/app/src/main/java/ru/dmkalvan/inote/MainActivity.java
package ru.dmkalvan.inote;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import ru.dmkalvan.inote.observe.Publisher;
import ru.dmkalvan.inote.ui.StartFragment;

public class MainActivity extends AppCompatActivity {

    private Navigation navigation;
    private final Publisher publisher = new Publisher();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        navigation = new Navigation(getSupportFragmentManager());
        initToolbar();
        getNavigation().addFragment(StartFragment.newInstance(), false);
    }

    private void initToolbar() {
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    public Navigation getNavigation() {
        return navigation;
    }

    public Publisher getPublisher() {
        return publisher;
    }
}

File: AndroidSDK/Kotlin/ContentProvider/app/src/main/java/ru/dmkalvan/contentprovider/MainActivity.kt
package ru.dmkalvan.contentprovider

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import ru.dmkalvan.contentprovider.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        if (savedInstanceState == null){
            supportFragmentManager.beginTransaction()
                .replace(R.id.container, ContactListFragment.newInstance())
                .commitAllowingStateLoss()
        }
    }

}